extmod/moducryptolib.c

```diff
@@ -119,6 +144,33 @@ STATIC void aes_process_cbc_impl(AES_CTX_IMPL *ctx, const uint8_t *in, uint8_t *
         AES_cbc_decrypt(ctx, in, out, in_len);
     }
 }
+
+#if MICROPY_PY_UCRYPTOLIB_CTR
+// axTLS doesn't have CTR support out of the box. This is based on `mbedtls_aes_crypt_ctr`.
+STATIC void aes_process_ctr_impl(AES_CTX_IMPL *ctx, const uint8_t *in, uint8_t *out, size_t in_len, struct ctr_params *ctr_params) {
```

Comment:
Would it be better to use the mbedtls CTR function if that's available (which might be faster than what's here and then could be shared with other uses, eg by TLS code), and have this explicit implementation only for axTLS?

Reply:
You're absolutely right, and that's exactly what happens now ;) This impl of `aes_process_ctr_impl` is only for axTLS. The mbedtls impl just calls `mbedtls_aes_crypt_ctr`.

Reply:
Ok great.  I missed the MICROPY_SSL_AXTLS guard above.